src/util.js

```diff
@@ -24,19 +24,23 @@ function ordinal(number) {
  * @param {string|null} originId origin channel id
  * @param {Discord.MessageEmbed} embed
  */
-async function sendEventLogMessage(guild, originId, embed) {
-	const logChannelId = db.getDB().get('channels.event-logs');
+async function sendEventLogMessage(logType, guild, originId, embed, file, components) {
+	const logChannelId = db.getDB().get(`${logType}`);
 	const logChannel = logChannelId && await guild.channels.fetch(logChannelId);
 	
 	const blacklistedIds = db.getDBList('channels.event-logs.blacklist');
 	if (blacklistedIds.includes(originId)) {
 		return;
 	}
-
 	if (!logChannel) {
 		console.log('Missing log channel!');
 	} else {
-		await logChannel.send({ embeds: [embed] });
+		// Check if there's an attached component
+		if (components) {
+			await logChannel.send({ embeds: [embed], files: [file], components: [components] });
+		} else {
+			await logChannel.send({ embeds: [embed], files: [file] });
+		}
```

Comment:
This check seems unnecessary? If `components` isn't sent then the array won't have any children, so it will function as if the array wasn't set at all

Reply:
This isn't how that works, it gets null, in which Discord.js will throw an error due to the fact that it simply cannot read properties of null. As weird that this check is, it is essential to make sure that it doesn't crash upon anything that isn't an edit message log.

Reply:
Amazing. I thought they actually sanity checked that data